### ISSUE TITLE
The example at in this section is not correct.

### ISSUE THREAD

**Issue by ekothe** [djnavarro/rbook#4]
See: https://github.com/djnavarro/rbook/blob/8aeeb95ba8729f6e4239033505c8449556490c29/bookdown/02.04-mechanics.Rmd#L701

The text from the PDF version is as follows 

> And besides, the problem that this causes is much more serious than a single sad nerd… because R has no way of knowing that the 1s in the group variable are a very different kind of thing to the 1s in the gender variable. So if I try to ask which elements of the group variable are equal to the corresponding elements in gender, R thinks this is totally kosher, and gives me this:
> ```
> group == gender
> [1] TRUE TRUE TRUE FALSE FALSE TRUE FALSE FALSE FALSE
> ```
> Well, that’s … especially stupid. The problem here is that R is very literal minded. Even though you’ve declared both group and gender to be factors, it still assumes that a 1 is a 1 no matter which variable it appears in.

But the example actually throws the following error

```
group == gender
## Error in Ops.factor(group, gender): level sets of factors are different
```

**Comment by djnavarro** [comment]
Yeah, this is definitely wrong. I swear there was some version of this where it did produce the output in the book even when the variables were factors, but I’m betting that what happened is that I changed some detail of the example and never noticed that this code snippet is invalid. (The big reason I’ve wanted to switch to R markdown is to fix all those errors)

**Comment by Maschette** [comment]
The other option would be to make gender have 3 variables (male, female, other/notspecified) and then the example works fine. 
''
> group <- c(1,1,1,2,2,2,3,3,3)
> group <- as.factor(group)
> gender<-c(3, 1, 1, 1, 1, 2, 2, 3,2 )
> gender<-as.factor(gender)
> group == gender
[1] FALSE  TRUE  TRUE FALSE FALSE  TRUE FALSE  TRUE FALSE''

**Comment by ekothe** [comment]
I like that solution @Maschette since it also has the side benefit of not implying that gender is dichotomous

**Comment by Maschette** [comment]
Yep that was one of the things I quite liked about it too.

**Comment by djnavarro** [comment]
Finally addressed the impact of changes in Ops.factor :)

Thankfully some of the other nuances about gender survived
